nightingale_body/src/nightingale_body/body_control.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python3
+
+import actionlib
+import rospy
+
+from moveit_action_handlers.msg import (
+    MoveToJointsMoveItAction,
+    MoveToJointsMoveItGoal,
+    PropertyValuePair,
+)
+
+from nightingale_msgs.srv import (
+    RobotConfigurationLookup,
+    RobotConfigurationLookupRequest,
+)
+
+
+class BodyJointControl:
+    def __init__(self):
+        self.head_ac = actionlib.simple_action_client.SimpleActionClient(
+            "/moveit_action_handlers/head/joint_ctrl", MoveToJointsMoveItAction
+        )
+        rospy.loginfo("Waiting for head joint moveit action server")
+        self.head_ac.wait_for_server()
+        rospy.loginfo("Found head joint moveit action server")
+
+        self.torso_ac = actionlib.simple_action_client.SimpleActionClient(
+            "/moveit_action_handlers/torso/joint_ctrl", MoveToJointsMoveItAction
+        )
+        rospy.loginfo("Waiting for torso joint moveit action server")
+        self.torso_ac.wait_for_server()
+        rospy.loginfo("Found torso joint moveit action server")
+
+    def _make_goal(self, jnt_name, jnt_pos, ee_vel=0.5, ee_accel=0.5, timeout=5):
+        goal = MoveToJointsMoveItGoal()
+
+        goal.endEffectorVelocity = ee_vel
+        goal.endEffectorAcceleration = ee_accel
+        goal.timeoutSeconds = timeout
+
+        for name, pos in zip(jnt_name, jnt_pos):
+            joint_pair = PropertyValuePair()
+            joint_pair.name = name
+            joint_pair.value = pos
+            goal.joint_pairs.append(joint_pair)
+        return goal
+
+    def cmd_head(self, target_jnt_name, target_jnt_pos):
+        goal = self._make_goal(target_jnt_name, target_jnt_pos)
+
+        self.head_ac.send_goal(goal)
+        return self.head_ac.wait_for_result()
+
+    def cmd_torso(self, target_jnt_name, target_jnt_pos):
+        goal = self._make_goal(target_jnt_name, target_jnt_pos)
+
+        self.torso_ac.send_goal(goal)
+        return self.torso_ac.wait_for_result()
+
+
+class BodyControl:
+    def __init__(self):
+        robot_config_service = "/nightingale/robot_configuration_lookup"
+        rospy.loginfo(f"Manipulation Control: waiting for robot configuration service")
+        rospy.wait_for_service(robot_config_service)
+        lookup_client = rospy.ServiceProxy(
+            robot_config_service, RobotConfigurationLookup
+        )
+        rospy.loginfo(
+            "Nightingale Manipulation Control found robot configuration lookup service server"
+        )
+        try:
+            response = lookup_client("home", RobotConfigurationLookupRequest.HEAD)
```

Comment:
Is it necessary to run these functions on init? I'm just scared something goes wrong and breaks the tablet holder. Also is this going to make the torso go high on init?

Reply:
No movement occurs on init, just looking up the values for the desired states